src/components/AgreementSections/InfoPeriods.js

```diff
@@ -0,0 +1,126 @@
+import React, { useEffect, useState } from 'react';
+import PropTypes from 'prop-types';
+import { FormattedMessage } from 'react-intl';
+import {
+  Button,
+  ButtonGroup,
+  Col,
+  FormattedUTCDate,
+  Headline,
+  KeyValue,
+  Layout,
+  NoValue,
+  Row,
+} from '@folio/stripes/components';
+
+export default function InfoPeriods({ periods }) {
+  /*
+   * This could be handled by callback, but since we need to know for display whether next/prev/current exist
+   * it seems more prudent to just fetch all three at once and then do our logic at this stage
+   */
+  const displayPeriodOptions = {
+    currentPeriod: periods?.find(p => p.periodStatus === 'current'),
+    nextPeriod: periods?.find(p => p.periodStatus === 'next'),
+    previousPeriod: periods?.find(p => p.periodStatus === 'previous')
+  };
```

Comment:
How does an example `periods` array look like? You are basically trying to convert an array to an object (Map) here. I would suggest we use `array.reduce` for this purpose instead of looping through each period here

Reply:
Hmmm, no. I'm doing no conversion, simply extracting the period p where p.periodStatus is equal to a certian value. This is what find is for, no?

array.reduce is for creating a value based on all the elements in the array, which I am not doing.


As per the proptypes:
```
periods: PropTypes.arrayOf(PropTypes.shape({
    cancellationDeadline: PropTypes.string,
    endDate: PropTypes.string,
    periodStatus: PropTypes.string,
    startDate: PropTypes.string.isRequired,
  })),
```
So we have an array of objects, each of which will have a periodStatus (Could be null).

That find command simply finds the one with a certain periodStatus, or returns undefined otherwise